crates/resolver/src/lookup.rs

```diff
@@ -184,14 +225,50 @@ impl<'a> Iterator for DnssecLookupRecordIter<'a> {
     }
 }
 
+// TODO: consider removing this as it's not a zero-cost abstraction
```

Comment:
same question w.r.t TODO w/o an issue. Blocker for merging? Follow-up work to be tracked?

(I'll stop explicitly flagging additional `TODO`/`FIXME`'s in my review since the feedback will be the same for each)